lib/read.js

```diff
@@ -63,8 +63,14 @@ function assembleReadQuery(opts) {
 
     var rels = compositionRelationCypherList(comps);
     var start = level ? '__sm_level' + (level - 1) : opts.varName;
+    var dir = (opts.compMapping[level].matches || opts.compMapping[level][rels.replace(/`/g, '')]).direction;
```

Comment:
I'm just not following how it works at all, to be honest. I don't understand where the `matches` property comes from; I don't see how the other side of the `||` will work if there is more than one composited property. Also I can't see how a read would work at all, using this method, if there was one composition with `direction: 'out'`, and one with `direction: 'in'`. I think the best approach is to not specify a direction at all in the read query, and deal with it later on in the `computeAndCoalesce` step (if you even need to deal with it at all? I'm not certain that you need to.)